ui/elyria/src/components/CustomerClaimForm.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { FormEvent, useState } from 'react';
+import Grid from '@material-ui/core/Grid';
+import Alert from '@material-ui/lab/Alert';
+import { TextField, Typography } from '@material-ui/core';
+import Button from '@material-ui/core/Button';
+import { useHistory } from 'react-router';
+import { useAuth } from '../hooks/useAuth';
+import useForm from '../hooks/useForm';
+import { claimCustomerData, searchCustomer } from '../services/c2dcrm';
+import SearchCustomersTable from './SearchCustomersTable';
+import { CrmListCustomersResponse } from '../services/c2dcrm.interface';
+
+export interface CustomerSearchForm {
+  email: string;
+  last_name: string;
+  house_number: string;
+}
+
+type CustomerClaimFormProps = {
+  onCustomerClaim: (claimedCustomer: CrmListCustomersResponse) => void;
+};
+
+const CustomerClaimForm: React.FC<CustomerClaimFormProps> = ({ onCustomerClaim }) => {
+  const history = useHistory();
+  const [customers, setCustomers] = useState<CrmListCustomersResponse[]>([]);
+  const { isAuthenticated, token } = useAuth();
+  const [error, setError] = useState('');
+  const { handleChange, inputs } = useForm<CustomerSearchForm>({
+    email: '',
+    last_name: '',
+    house_number: '',
+  });
+
+  if (!isAuthenticated) history.push('/');
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    try {
+      setError('');
+      const res = await searchCustomer(token, inputs);
+
+      if (res.data.length > 0) {
+        setCustomers(res?.data);
+      } else {
+        setError('Records with such details not found.');
+        setCustomers([]);
+      }
+    } catch (e) {
+      setCustomers([]);
+      setError('Something went wrong. Please try again.');
+    }
+  };
+
+  const handleCustomerClaim = async (customer: CrmListCustomersResponse) => {
+    try {
+      setError('');
+      const res = await claimCustomerData(token, { id: customer.id });
+
+      if (res.data) {
+        onCustomerClaim(res.data);
+      } else {
+        setCustomers([]);
```

Comment:
If the claim fails or doesn't get the expect response, you reset the customers - is this the expected outcome?

Reply:
Hmm probably not! Good spot!